### ISSUE TITLE
Idea regarding the removed testing code: use/upload legal videos

### ISSUE THREAD

**Issue by solbu** [l1ving/youtube-dl#5]
Hi.
It seems that the original repo was taken down because it explisitly used non-free content from Youtube in it's test code, and RIAA used this as the excuse to take down youtube-dl.

An idea came to me while reading the code in question.
Someone that has a Youtube account could upload videos that one owns the copyrights to, and thus can license under the Creativ Comons license that youtube allows for.
So one can have one open video, one age restricted video and what else one need to test against.

Then one can have test videos to use which RIAA and other copyright agencies cannot legaly file DMCA complaint against.

**Comment by l1ving** [comment]
That was my original intention but I don't have replacement videos for the specific tests that were removed. I guess it's a scavenger hunt now :p

**Comment by kisik21** [comment]
Maybe use a Creative Commons or a public domain video? I see Big Buck Bunny all over the place when evaluating performance of encoders, maybe we can leverage that...

**Comment by l1ving** [comment]
Yep, I'd love to switch all the tests to videos licensed under CC0 but I don't know the specifics of how to find videos that fit the test requirements.

**Comment by mphirke** [comment]
Youtube has a filter for Creative Commons right underneath the search so finding videos wouldn't be too much trouble, at least on YouTube. 

![image](https://user-images.githubusercontent.com/22197137/97158324-43394e00-179f-11eb-899f-23aa373b847c.png)

But finding Creative Commons content on FB/Vimeo/YouPorn/tumblr/PBS will be difficult. I don't think it's possible to even get Creative Commons content on PBS because videos uploaded there are licensed to PBS so to keep this repo legal, the entire PBS part may have to be omitted. 

For YouTube playlists, I'm not entirely sure how it goes legally but I think it would make sense to think a collection of creative commons videos into a playlist would(should?) also be under creative commons. I'll try to compile a few links anyway.

**Comment by solbu** [comment]
> But finding Creative Commons content on FB/Vimeo/YouPorn/tumblr/PBS will be difficult. I don't think it's possible to even get Creative Commons content on PBS because videos uploaded there are licensed to PBS so to keep this repo legal, the entire PBS part may have to be omitted.

The part of the testcode that requires questionable download tests doesn't need to be included in the public available repo. It can be moved to it's own test file and exclude that file using «.gitignore».

Also, I don't understand why the name of the video need to be in the test code. Somehow i get the feeling that this could all been avoided if the «Icona Pop» link was not identified by name, as I suspect the DMCA came because the name was searchable. All one really should need is the url and a description of what is being tested.

**Comment by solbu** [comment]
> Somehow i get the feeling that this could all been avoided if the «Icona Pop» link was not identified by name

Should read … «avoided if for example the «Icona Pop» link…»

**Comment by calvinthefreak** [comment]
May use the Videos from CopyMe: (https://www.youtube.com/watch?v=XXxzWgl3nHs; https://www.youtube.com/watch?v=FMo6e7VXKkg) Maybe that will be a good Point for the big asses out there....

The funniest thing in their videos is: 
![grafik](https://user-images.githubusercontent.com/5560381/97237847-4c9ad880-17e8-11eb-8d17-6310db85f596.png)

**Comment by calvinthefreak** [comment]
Ah, and they also have:
https://vimeo.com/copymetv


Maybe just search for anything, that has some Creative Commons License...
I know, some sites like YP may be difficult... but I think there is still creative commons stuff on there...
Or just Upload something, that Says: "Test-Video" or so... and let it be checked...?

**Comment by l1ving** [comment]
If something on youtube is Creative Commons 0 then we can reupload onto whichever site needs to be tested.

**Comment by mrwesleycrusher** [comment]
TLG just gave rights to all his past and current youtube videos for the project: https://www.youtube.com/watch?v=lU0GtLrSLe8

**Comment by ZenulAbidin** [comment]
Don't forget that each test case that had to be removed was testing a specific feature of youtube-dl, so each replacement video used as an alternative must have the same technical properties as one of the infringing videos. I mean e.g. the age-limited video test, the geoblock test, the Featured videos test, and others, each of the videos that we use as replacements needs to have one of these properties.

Using freely available videos is good but I don't see the point in using several of them if they all are testing if video grabbing works and no edge cases are being tested.

Don't forget that we literally have a Vevo extractor too, so the RIAA is probably going to target that next. But we should leave that up too (using the same reasoning as #9)

**Comment by Rohitrajak1807** [comment]
Try this: https://www.youtube.com/watch?v=lU0GtLrSLe8
The Linux Gamer(Gardiner Bryant) could help with this issue

**Comment by solbu** [comment]
> The Linux Gamer(Gardiner Bryant) could help with this issue

For TLG to help with this he need to upload an Age restricted test video and a region restricted test video.
Or someone else having a youtube account need to do so and give us permission to use them.

**Comment by ZenulAbidin** [comment]
I just uploaded an age-restricted video under Creative Commons Attribution license (It's just Big Buck Bunny): https://youtu.be/PA4gbtKWNAI

I can't upload a region restricted video because apparently the option to do that is only available for "Content Partners using ContentID" ([support link](https://support.google.com/youtube/answer/6303378?hl=en))

We need to find a video used by a content partner that's *not* obvious copyrighted material. Pretty much everything uses the Standard Youtube License, but we'd need to stay away from music videos and everything else inside the RIAA's domain (I don't think they have control over political videos for example)

**Comment by l1ving** [comment]
Sounds good, thank you! Remind me by tomorrow if I haven't readded the age restricted test

**Comment by solbu** [comment]
> We need to find a video used by a content partner that's _not_ obvious copyrighted material.

The «test» is _not_ whether something is copyrighted or not, but whether it is under a Free license.
Big Buck Bunny is copyrighted. If it was not copyrighted, there would not be any need for a CC license.
Even the Linux kernel is copyrighted, but it is given a license that even the copyright holder can't take away from you.

**Comment by ultrasound1372** [comment]
I am an independent artist who just uploaded my music through DistroKid and got that on YouTube through a topic channel, I own the copyrights because I wrote the songs and DistroKid doesn't take that away from me, I'm not sure if I'm under the Umbrella of the RIAA because I'm not signed with a record label. It's not a VEVO but it's something similar I think. Also, for political videos that may be geo restricted try something like C-Span?

**Comment by ZenulAbidin** [comment]
> Sounds good, thank you! Remind me by tomorrow if I haven't readded the age restricted test

I don’t see it committed yet, but when I get home today I’ll probably put that test in myself and submit a PR for it.

**Comment by l1ving** [comment]
Sorry, I got caught up in other work. I'd appreciate a PR if you'd like to do it, thanks
